### PR TITLE
deleters of service and service-role

### LINKED ISSUE #1155: Add a space-cli delete <resource> command
### The problem faced currently?
Currently we can apply config to SpaceCloud. This solves the need to create and update resources. However, there is no mechanism to delete a resource via. space-cli.
### How can we solve it?
We can add `deleters` for each resource. This will be similar to the `generators` and `getters`. We will have to be careful to make sure we give flexibility to delete resources by `id`, `prefix` (this can be a multi select) or delete all resources.
>If you want this feature to be implemented, give it a thumbs up reaction, so that we can determine which features are important to you.
>👍

### PR DESCRIPTION
**Pull request by prithvi081099**
Fix #1155 and #1462

### CODE REVIEW

**Review by sharadregoti — CHANGES_REQUESTED** [review]

File: space-cli/cmd/modules/services/delete_test.go [hunk 1]
```diff
@@ -528,3 +528,1065 @@ func Test_deleteSecret(t *testing.T) {
...
+							Options: []string{"localAdmin::v1", "local::v1"},
+							Default: []string{"localAdmin::v1", "local::v1"}[0],
+						},
+						&surveyNoMatchReturnValue,
+					},
+					paramsReturned: []interface{}{nil, "localAdmin::v1"},
+				},
+			},
+		},
+	}
```
Comment (sharadregoti) [inline space-cli/cmd/modules/services/delete_test.go:1037]: Add some test cases for 

- Exact match
- User has provided both service id & version
- Add more services in, Get services HTTP call

File: space-cli/cmd/modules/services/delete_test.go [hunk 2]
```diff
@@ -528,3 +528,1065 @@ func Test_deleteSecret(t *testing.T) {
...
+		prefix  map[string]string
+	}
+	tests := []struct {
+		name              string
+		args              args
+		transportMockArgs []mockArgs
+		surveyMockArgs    []mockArgs
+		wantErr           bool
+	}{
+		// TODO: Add test cases.
```
Comment (sharadregoti) [inline space-cli/cmd/modules/services/delete_test.go:1084]: Remove this & do the same here

**Review by sharadregoti — CHANGES_REQUESTED** [review]

**Review by sharadregoti — APPROVED** [review]

File: space-cli/cmd/modules/services/commands.go [hunk 3]
```diff
@@ -38,64 +38,15 @@ func GenerateSubCommands() []*cobra.Command {
...
-			if err != nil {
-				return nil, cobra.ShellCompDirectiveDefault
-			}
-			var ids []string
-			for _, v := range obj {
-				ids = append(ids, v.Meta["id"])
-			}
-			return ids, cobra.ShellCompDirectiveDefault
-		},
+		Use:               "service-routes",
```
Comment (YourTechBud) [inline space-cli/cmd/modules/services/commands.go:41]: Plural form must be in alias

File: space-cli/cmd/modules/services/commands.go [hunk 4]
```diff
@@ -38,64 +38,15 @@ func GenerateSubCommands() []*cobra.Command {
...
-				}
-				var roleID []string
-				for _, v := range objs {
-					roleID = append(roleID, v.Meta["roleId"])
-				}
-				return roleID, cobra.ShellCompDirectiveDefault
-			}
-			return nil, cobra.ShellCompDirectiveDefault
-		},
+		Use:               "service-role",
```
Comment (YourTechBud) [inline space-cli/cmd/modules/services/commands.go:47]: Plural form must be in alias

File: space-cli/cmd/modules/services/commands.go [hunk 5]
```diff
@@ -105,43 +56,9 @@ func GetSubCommands() []*cobra.Command {
...
-				}
-				var versions []string
-				for _, v := range objs {
-					versions = append(versions, v.Meta["version"])
-				}
-				return versions, cobra.ShellCompDirectiveDefault
-			}
-			return nil, cobra.ShellCompDirectiveDefault
-		},
+		Use:               "services",
```
Comment (YourTechBud) [inline space-cli/cmd/modules/services/commands.go:59]: Plural form must be in alias

File: space-cli/cmd/modules/services/generate.go [hunk 6]
```diff
@@ -46,9 +46,12 @@ func GenerateService(projectID, dockerImage string) (*model.SpecObject, error) {
 		}
 		registry, present := p[0].Spec.(map[string]interface{})["dockerRegistry"]
 		if registry == "" || !present {
-			return nil, fmt.Errorf("no docker registry provided for project (%s)", projectID)
+			reg := ""
+			if err := input.Survey.AskOne(&survey.Input{Message: "No default registry configured, provide registry"}, &reg); err != nil {
```
Comment (YourTechBud) [inline space-cli/cmd/modules/services/generate.go:50]: Add an example `gcr.io/spaceuptech`

**Review by YourTechBud — CHANGES_REQUESTED** [review]

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**